transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -163,7 +163,15 @@ public void run() {
     private void invokeChannelRegistered() {
         if (invokeHandler()) {
             try {
-                ((ChannelInboundHandler) handler()).channelRegistered(this);
+                final ChannelHandler handler = handler();
+                // DON'T CHANGE
+                // Duplex handlers implements both out/in interfaces causing a scalability issue
+                // see https://bugs.openjdk.org/browse/JDK-8180450
+                if (handler instanceof ChannelDuplexHandler) {
```

Comment:
Instead of placing this 3 line comment everywhere, we can create an issue with all details and put its link in a single comment. It will be clean and exact.

Reply:
Good point, but can still happen to accidentally remove them without doing git blame :( (thinking about future)

Reply:
Lets just keep it... Better safe than sorry.